stylelint-polaris/plugins/coverage/index.js

```diff
@@ -1,7 +1,445 @@
 const stylelint = require('stylelint');
+const {
+  createVar,
+  getThemeVarNames,
+  themeDefault,
+} = require('@shopify/polaris-tokens');
 
 const {isPlainObject} = require('../../utils');
 
+const objectOf = (keys, value) =>
+  keys.reduce((memo, key) => ({...memo, [key]: value}), {});
+
+const matchNameRegExp = (name) =>
+  // Using `^` to match the start of a string since postcss normalizes name properties
+  // https://regex101.com/r/3tzvIW/1
+  new RegExp(String.raw`^([\w-]+\.)?(?<![\w-])${name}(?![\w-])`);
+
+const disallowedUnits = [
+  'px',
+  'rem',
+  'em',
+  '%',
+  'ex',
+  'ch',
+  'lh',
+  'rlh',
+  'vw',
+  'vh',
+  'vmin',
+  'vmax',
+  'vb',
+  'vi',
+  'svw',
+  'svh',
+  'lvw',
+  'lvh',
+  'dvw',
+  'dvh',
+  'cm',
+  'mm',
+  'Q',
+  'in',
+  'pc',
+  'pt',
+];
+
+/**
+ * @type {import('./plugins/coverage').PrimaryOptions} The stylelint-polaris/coverage rule expects a 3-dimensional rule config that groups Stylelint rules by coverage categories. It reports problems with dynamic rule names by appending the category to the coverage plugin's rule name
+
+(e.g., Unexpected named color "blue" - Please use a Polaris color token Stylelint(polaris/colors/color-named)")
+*/
+const stylelintPolarisCoverageOptions = {
```

Comment:
Can/should the rule's configuration move into its own file instead of into the rule's plugin file?

Reply:
Yeah, definitely can - I just wholesale copy+pasted it from the stylelint-polaris `index` file 🤷